### PR TITLE
Typescript

### PR DESCRIPTION
**Pull request by vincaslt**
Ready for review and merge

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -99,13 +104,15 @@
...
+    "prettier": "^1.18.2",
+    "typescript": "^3.6.3"
   },
   "scripts": {
     "lint": "eslint $(find cli bin -type f) --quiet --cache",
     "pretty": "prettier --write  cli bin",
     "preversion": "git update-index --no-assume-unchanged package-lock.json",
-    "postversion": "git update-index --assume-unchanged package-lock.json && git push --tags && git push"
+    "postversion": "git update-index --assume-unchanged package-lock.json && git push --tags && git push",
+    "check-types": "tsc"
```
Comment (vincaslt) [inline package.json]: should probably use `--noEmit` flag, we don't actually want typedef files. Also, probably useless in this project, but the idea is to have type checking before commit.

Comment (dhoko) [reply package.json]: proton-pack won't be in TS ;)

File: package.json [hunk 2]
```diff
@@ -21,8 +21,13 @@
     "@babel/plugin-transform-runtime": "^7.6.2",
     "@babel/preset-env": "^7.6.2",
     "@babel/preset-react": "^7.0.0",
+    "@babel/preset-typescript": "^7.3.3",
     "@babel/runtime": "^7.6.2",
     "@hot-loader/react-dom": "^16.9.0",
+    "@types/react": "^16.8.22",
```
Comment (vincaslt) [inline package.json]: Not needed for this project.

**Review by mmso — APPROVED** [review]

**Comment by vincaslt** [comment]
woohoo

**Pull request merged**